### PR TITLE
Docufixes

### PR DESCRIPTION
**Pull request by sbird**
This PR adds documentation strings to all the parameters that were lacking them. It also makes optional some parameters where a default can be worked out from the ICs.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -99,36 +96,30 @@ In the end, we will have 2 binaries:
...
-GLIBC 2.22
-----------
+Config Files
+------------
 
-Cray updated their GLIBC to 2.22+ recently. 
-A good move but it happens to be a buggy version of GLIBC:
-https://sourceware.org/bugzilla/show_bug.cgi?id=19590
-causing non-existing symbols like `_ZGVcN4v___log_finite`.
-Adding `-lmvec -lmvec_nonshared` to GSL_LIBS works around the issue.
```
Comment (rainwoodman) [inline README.rst]: The `-lmvec -lmvec_nonshared` workaround is very likely still relevant.

Comment (rainwoodman) [reply README.rst]: Ah it is simply moved.

File: gadget/params.c [hunk 2]
```diff
@@ -210,18 +213,19 @@ create_gadget_parameter_set()
...
+    param_declare_int(ps, "RadiationOn", OPTIONAL, 1, "Include radiation density in the background evolution.");
     param_declare_int(ps, "FastParticleType", OPTIONAL, 2, "Particles of this type will not decrease the timestep. Default neutrinos.");
 
     param_declare_double(ps, "GravitySoftening", OPTIONAL, 1./30., "Softening for collisionless particles; units of mean separation of DM. ForceSoftening is 2.8 times this.");
     param_declare_double(ps, "GravitySofteningGas", OPTIONAL, 1./30., "Softening for collisional particles (Gas); units of mean separation of DM; 0 to use Hsml of last step. ");
 
     param_declare_int(ps, "ImportBufferBoost", OPTIONAL, 6, "Memory factor to allow for there being more particles imported during treewlk than exported. Increase this if code crashes during treewalk with out of memory.");
-    param_declare_double(ps, "PartAllocFactor", REQUIRED, 0, "Over-allocation factor of particles, to deal with load imbalances.");
+    param_declare_double(ps, "PartAllocFactor", OPTIONAL, 1.5, "Over-allocation factor of particles. The load can be imbalanced to allow for the work to be more balanced.");
```
Comment (rainwoodman) [inline gadget/params.c]: Usually higher resolution runs need larger values.

File: gadget/params.c [hunk 3]
```diff
@@ -231,23 +235,23 @@ create_gadget_parameter_set()
...
+    param_declare_int(ps, "FOFHaloMinLength", OPTIONAL, 32, "Minimum number of particles per FOF Halo.");
     param_declare_double(ps, "MinFoFMassForNewSeed", OPTIONAL, 5e2, "Minimal Mass for seeding tracer particles ");
     param_declare_double(ps, "TimeBetweenSeedingSearch", OPTIONAL, 1e5, "Time Between Seeding Attempts: default to a a large value, meaning never.");
 
     /*Black holes*/
     param_declare_int(ps, "BlackHoleOn", REQUIRED, 1, "Enable Blackhole ");
-    param_declare_double(ps, "BlackHoleAccretionFactor", OPTIONAL, 100, "");
-    param_declare_double(ps, "BlackHoleEddingtonFactor", OPTIONAL, 3, "");
-    param_declare_double(ps, "SeedBlackHoleMass", OPTIONAL, 5e-5, "");
+    param_declare_double(ps, "BlackHoleAccretionFactor", OPTIONAL, 100, "BH accretion rate as a fraction of the Bondi accretion rate.");
```
Comment (rainwoodman) [inline gadget/params.c]: BH accretion boosting factor relative to the rate from the Bondi accretion model.

File: gadget/params.c [hunk 4]
```diff
@@ -231,23 +235,23 @@ create_gadget_parameter_set()
...
+    param_declare_double(ps, "BlackHoleEddingtonFactor", OPTIONAL, 3, "Maximum Black hole accretion as a function of Eddington.");
+    param_declare_double(ps, "SeedBlackHoleMass", OPTIONAL, 5e-5, "Minimal black hole seed mass in internal mass units.");
```
Comment (rainwoodman) [inline gadget/params.c]: Mass of initial black hole seed in internal mass units.

File: gadget/params.c [hunk 5]
```diff
@@ -231,23 +235,23 @@ create_gadget_parameter_set()
...
 
-    param_declare_double(ps, "BlackHoleNgbFactor", OPTIONAL, 2, "");
+    param_declare_double(ps, "BlackHoleNgbFactor", OPTIONAL, 2, "Factor by which to increase the number of neighbours for a black hole.");
 
-    param_declare_double(ps, "BlackHoleMaxAccretionRadius", OPTIONAL, 99999., "");
-    param_declare_double(ps, "BlackHoleFeedbackFactor", OPTIONAL, 0.05, "");
-    param_declare_double(ps, "BlackHoleFeedbackRadius", OPTIONAL, 0, "");
+    param_declare_double(ps, "BlackHoleMaxAccretionRadius", OPTIONAL, 99999., "Maximum neighbour search radius for black holes.");
```
Comment (rainwoodman) [inline gadget/params.c]: ; rarely need to set this limit.

File: gadget/params.c [hunk 6]
```diff
@@ -231,23 +235,23 @@ create_gadget_parameter_set()
...
+    param_declare_double(ps, "BlackHoleFeedbackFactor", OPTIONAL, 0.05, " Fraction of the black hole luminosity to turn into thermal energy");
+    param_declare_double(ps, "BlackHoleFeedbackRadius", OPTIONAL, 0, "Maximum comoving (if > 0) radius at which the black hole feedback energy is deposited.");
```
Comment (rainwoodman) [inline gadget/params.c]: If set, the comoving radius at which the blackhole ...

File: gadget/params.c [hunk 7]
```diff
@@ -231,23 +235,23 @@ create_gadget_parameter_set()
...
 
-    param_declare_double(ps, "BlackHoleFeedbackRadiusMaxPhys", OPTIONAL, 0, "");
+    param_declare_double(ps, "BlackHoleFeedbackRadiusMaxPhys", OPTIONAL, 0, "Maximum physical (if > 0) radius at which the black hole feedback energy is deposited.");
```
Comment (rainwoodman) [inline gadget/params.c]: If set, the physical .... When both this flag and BlackHelFeedbackRadius are both set, the smaller radius is used.

File: gadget/params.c [hunk 8]
```diff
@@ -262,51 +266,54 @@ create_gadget_parameter_set()
...
+        {"continuous", SFR_CRITERION_CONTINUOUS_CUTOFF}, /* Modify self-gravitating star formation to smooth the star formation threshold. From Phil Hopkins.*/
         {NULL, SFR_CRITERION_DENSITY},
     };
 
     static ParameterEnum WindModelEnum [] = {
-        {"subgrid", WIND_SUBGRID},
-        {"decouple", WIND_DECOUPLE_SPH},
-        {"halo", WIND_USE_HALO},
-        {"fixedefficiency", WIND_FIXED_EFFICIENCY},
+        {"subgrid", WIND_SUBGRID}, /* the original model of SH03, in which winds are included by returning energy to the star forming regions. This cools away and is ineffective.*/
```
Comment (rainwoodman) [inline gadget/params.c]: the vanilla model proposed in SH03.

File: gadget/params.c [hunk 9]
```diff
@@ -262,51 +266,54 @@ create_gadget_parameter_set()
...
+        {"decouple", WIND_DECOUPLE_SPH}, /* Specifies that wind particles are created temporarily decoupled from the gas dynamics */
+        {"halo", WIND_USE_HALO}, /* Wind speeds depend on the halo circular velocity*/
+        {"fixedefficiency", WIND_FIXED_EFFICIENCY}, /* Winds have a fixed efficiency and thus fixed wind speed*/
         {"sh03", WIND_SUBGRID | WIND_DECOUPLE_SPH | WIND_FIXED_EFFICIENCY} ,
```
Comment (rainwoodman) [inline gadget/params.c]: The canonical SH03 model.

File: gadget/params.c (hunk 2 above)
Comment (sbird) [reply gadget/params.c]: Ah but not really when one is running with a lot of threads. In that case the particle-load balance is rarely above 1.1.

**Pull request merged**